yt/data_objects/tests/test_data_containers.py

```diff
@@ -0,0 +1,121 @@
+import shelve
+
+import numpy as np
+from nose.tools import assert_raises
+from numpy.testing import assert_array_equal
+
+from yt.data_objects.data_containers import YTDataContainer
+from yt.testing import assert_equal, fake_random_ds, fake_amr_ds, fake_particle_ds
+from yt.utilities.exceptions import YTFieldNotFound
+
+
+def test_yt_data_container():
+    # Test if ds could be None
+    with assert_raises(RuntimeError) as err:
+        YTDataContainer(None, None)
+    desired = 'Error: ds must be set either through class type or parameter' \
+              ' to the constructor'
+    assert_equal(str(err.exception), desired)
+
+    # Test if field_data key exists
+    ds = fake_random_ds(5)
+    proj = ds.proj("density", 0, data_source=ds.all_data())
+    assert_equal('px' in proj.keys(), True)
+    assert_equal('pz' in proj.keys(), False)
+
+    # Delete the key and check if exits
+    proj.__delitem__('px')
+    assert_equal('px' in proj.keys(), False)
+    proj.__delitem__('density')
+    assert_equal('density' in proj.keys(), False)
+
+    # Delete a non-existent field
+    with assert_raises(YTFieldNotFound) as ex:
+        proj.__delitem__('p_mass')
+    desired = "Could not find field '('stream', 'p_mass')' in UniformGridData."
+    assert_equal(str(ex.exception), desired)
+
+    # Test the convert method
+    assert_equal(proj.convert('HydroMethod'), -1)
```

Comment:
Don't delete `__delitem__`, that's still needed. I was saying that *in the test* don't invoke `__delitem__` directly, instead use the `del` statement to invoke it indirectly.

You should delete `convert` since it's a relic function that doesn't do what it says in its own docstring.